components/Starknet/modules/architecture-and-concepts/pages/network-architecture/messaging-mechanism.adoc

```diff
@@ -207,12 +207,13 @@ The hash of the corresponding L1 handler transaction on L2 is computed as follow
 
 [source,cairo]
 ----
-l1_handler_tx_hash = ℎ(
+l1_handler_tx_hash = _h_(
     "l1_handler",
     version,
     contract_address,
     entry_point_selector,
-    ℎ(calldata),
+    _h_(calldata),
+    0,
```

Comment:
```suggestion
    0, // L1 to L2 message fees are charged on the L1.
```

Reply:
@glihm Good clarification. I moved this to the list of explanations below.